### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by boricj**
This is a fairly straightforward CRUD+eXecute endpoint for user scripts as part of the automation project. The actual script for running things under `_bios-script` will appear in a soon-to-be-created pull request for fty-core, as this utility may be useful beyond this applet.

### CODE REVIEW

**Comment by boricj** [comment]
Related pull request at fty-core created (https://github.com/42ity/fty-core@294).

File: project.xml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    </target>
+
+    <classfilename keep-tree = "true" use-cxx = "true" pretty-print = "no" source-extension = "cc" />
+    <include filename = "license.xml" />
+    <version major = "1" minor = "0" patch = "0" />
+    <abi current = "1" revision = "0" age = "0" />
+
+    <use project = "cxxtools" test="cxxtools::Utf8Codec::Utf8Codec" header="cxxtools/allocator.h"
+        repository = "https://github.com/42ity/cxxtools.git"
+        release = "release/IPM_Infra-1.4"
```
Comment (geraldguillaume) [inline project.xml]: Why using 1.4 branch ?

File: project.xml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        />
+
+    <use project = "tntnet"
+        header = "tnt/tntnet.h"
+        repository = "https://github.com/42ity/tntnet.git"
+        release = "release/IPM_Infra-1.4"
```
Comment (geraldguillaume) [inline project.xml]: Why using 1.4 branch ?

File: src/rest_scripts_execute_POST.ecpp [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+    argv.push_back(checked_name.c_str());
+    for (const auto &i : parameters) {
+        argv.push_back(i.c_str());
+    }
+    argv.push_back(nullptr);
+
+    // Launch child.
+    gettimeofday(&tvStart, nullptr);
+
+    if (posix_spawnp(&childPid, "/usr/bin/sudo", &fileActions, nullptr, (char* const*)&argv[0], environ) != 0) {
```
Comment (geraldguillaume) [inline src/rest_scripts_execute_POST.ecpp]: MlmSubprocess class was developped for this goal.
Could you use it in place of your posix_spawnp approach

**Comment by geraldguillaume** [comment]
remove  CMakeLists.txt

File: src/rest_scripts_execute_POST.ecpp (hunk 3 above)
Comment (boricj) [reply src/rest_scripts_execute_POST.ecpp]: After checking out MlmSubprocess, it does not offer a way to merge STDOUT and STDERR into one stream which is what we want for user scripts. Its implementation must be improved in this regard in order to switch to it.

**Comment by boricj** [comment]
Note: pull request now depends on https://github.com/42ity/fty-common-mlm/pull/20

File: project.xml [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    license = "GPL-2.0+"
+    repository = "https://github.com/42ity/fty-scripts-rest"
+    >
+
+    <!-- Note: Eaton proprietary components do not make Travis recipes,
+         but do store and (in other teams) use the CMake build recipes -->
+    <target name = "debian" />
+    <target name = "redhat" />
+    <target name = "jenkins" >
+        <option name = "agent_label" value = "devel-image-ipm_1.4 &amp;&amp; x86_64" />
```
Comment (geraldguillaume) [inline project.xml]: do not use 1.4 image.
set devel-image

**Pull request merged**